### PR TITLE
Fixes #379 Mockito.after(..) doesn't verify in a loop anymore

### LINKED ISSUE #379: Verification using After and ArgumentCaptor with Times or AtLeast methods returns unexpected size of captured values list.
Hi!
I was looking issue #345 and found that the test "shouldReturnListOfArgumentsWithSameSizeAsGivenInAtMostVerification" (located in VerificationAfterDelayTest.java) fails if you use atLeast() or times() instead of atMost().
I created these tests:

screenshot from 2016-03-24 16 50 29

### PR DESCRIPTION
**Pull request by ChristianSchwarz**
Fixes #379

The implementation of Mockito.after(..) doesn't loop anymore but waits until the specified time is elapsed, as specified in the JavaDoc of `Mockito.after(long)`.

### CODE REVIEW

File: src/test/java/org/mockitousage/verification/VerificationAfterDelayTest.java [hunk 1]
```diff
@@ -4,152 +4,190 @@
...
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Captor;
 import org.mockito.Mock;
 import org.mockito.exceptions.base.MockitoAssertionError;
-import org.mockitoutil.TestBase;
-
-public class VerificationAfterDelayTest extends TestBase {
```
Comment (TimvdLippe) [inline src/test/java/org/mockitousage/verification/VerificationAfterDelayTest.java]: Why did you reformat this file? I think the previous version was more readible and clear for me. Also the indentation seems to be off.

**Comment by TimvdLippe** [comment]
Could you rebase+squash and update the coverage so the codecov check does not fail? Also there was a previous PR solving the same issue (#380). Should we maybe check which one is better?

**Comment by ChristianSchwarz** [comment]
@TimvdLippe Thanks for reviewing this PR.I completely overlooked  #380, it is not referenced in issue #379. 
This PR fixes the `after(..)` implementation by waiting the specified time and verifing it afterwards no polling is needed. The implementation follows the JavaDoc of Mockito.after(long):
> Allows verifying over a given period. It causes a verify to **wait for a specified period of time** for a desired interaction

On the other hand I learned that e.g: Mockito.after(long).atMost() must fail fast, that behaviour is not implemented in this PR. Currently we have no tests for fail-fast behaviour in the master-branch that should be fixed first.

After working on this PR I think we should re-think the current API for concurrent invocation verification. I see no reason to have both `after(long)` and `timeout(long)` but that should be discussed in an other issue.

**Comment by ChristianSchwarz** [comment]
Should I close this PR?

**Comment by TimvdLippe** [comment]
I think it is wise to reach a concensus in the issue first, but in the mean time we can keep the PRs open. I will mark it as blocked for resolution of #379.

**Comment by raphw** [comment]
Can we close this ticket since #379 is closed?

**Comment by bric3** [comment]
Yep that makes sense

**Pull request closed without merging**